### PR TITLE
feat(cli): added decompression benchmark

### PR DESCRIPTION
**Pull request by jkowalski**

### CODE REVIEW

File: cli/command_benchmark_compression.go [hunk 1]
```diff
@@ -155,11 +212,86 @@ func (c *commandBenchmarkCompression) run(ctx context.Context) error {
 			return compressedSize
 		}
 
+		outputBuffers := makeOutputBuffers(c.parallel, defaultCompressedDataByMethod)
```
Comment (julio-lopez) [inline cli/command_benchmark_compression.go:215]: What prevents GC from running while benchmarking and affecting the results? Is GC turned off? would the last method to be benchmarked be likely to experience more memory pressure?

File: cli/command_benchmark_compression.go [hunk 2]
```diff
@@ -106,13 +128,49 @@ func (c *commandBenchmarkCompression) run(ctx context.Context) error {
...
+		}
+	}
+
+	return nil
+}
+
+func (c *commandBenchmarkCompression) runCompression(ctx context.Context, data []byte, repeatCount int, algorithms map[compression.Name]compression.Compressor) error {
+	var results []compressionBechmarkResult
+
+	log(ctx).Infof("Compressing input file %q (%v) using all compression methods.", c.dataFile, units.BytesString(int64(len(data))))
```
Comment (julio-lopez) [inline cli/command_benchmark_compression.go:171]: is it still the case that _all_ compression methods are used here?

File: cli/command_benchmark_compression.go [hunk 3]
```diff
@@ -82,8 +88,26 @@ type compressionBechmarkResult struct {
...
+	if c.algorithms == "" {
+		if compression.IsDeprecated[name] && !c.deprecated {
+			return false
+		}
+
+		return true
+	}
+
+	for _, a := range strings.Split(c.algorithms, ",") {
+		if strings.HasPrefix(string(name), a) {
```
Comment (julio-lopez) [inline cli/command_benchmark_compression.go:101]: for my understanding: why `HasPrefix`?

File: cli/command_benchmark_compression.go [hunk 4]
```diff
@@ -106,13 +128,49 @@ func (c *commandBenchmarkCompression) run(ctx context.Context) error {
...
+	}
+
+	if benchmarkCompression {
+		if err := c.runCompression(ctx, data, repeatCount, algorithms); err != nil {
+			return err
 		}
+	}
+
+	if benchmarkDecompression {
+		if err := c.runDecompression(ctx, data, repeatCount, algorithms); err != nil {
```
Comment (julio-lopez) [inline cli/command_benchmark_compression.go:160]: How does this work when multiple algorithms are specified and only decompression is benchmarked? essentially, what's the input for multiple decompressors when there is no prior compression.

**Review by julio-lopez — COMMENTED** [review]
+1: LG 👍

A few comments and questions.
Feel free to merge, no need for an additional round of review here.

Thanks.

File: cli/command_benchmark_compression.go (hunk 3 above)
Comment (jkowalski) [reply cli/command_benchmark_compression.go:101]: so you can test all "zstd" in one shot

File: cli/command_benchmark_compression.go (hunk 4 above)
Comment (jkowalski) [reply cli/command_benchmark_compression.go:160]: we compress once per compressor, then decompress N times.

File: cli/command_benchmark_compression.go (hunk 1 above)
Comment (jkowalski) [reply cli/command_benchmark_compression.go:215]: we actually call `runtime.GC()` before each algorithm.

File: cli/command_benchmark.go [hunk 5]
```diff
@@ -30,30 +31,58 @@ type cryptoBenchResult struct {
...
 			defer wg.Done()
 
-			run()
+			run(arg)
 		}()
 	}
 
 	// run one on the main goroutine and N-1 in parallel.
-	v := run()
+	v := run(args[len(args)-1])
```
[the remark this answers is not in the document]
Comment (jkowalski) [reply cli/command_benchmark.go:73]: premature optimization :)

File: cli/command_benchmark_compression.go (hunk 4 above)
Comment (julio-lopez) [reply cli/command_benchmark_compression.go:160]: I see. I missed that. Thanks.

File: cli/command_benchmark_compression.go (hunk 2 above)
Comment (jkowalski) [reply cli/command_benchmark_compression.go]: replaced "all" with a number.

**Review by julio-lopez — APPROVED** [review]
+1: LG 👍

**Pull request merged**